keps/sig-network/1111-dynamically-re-configure-lb-node-set/README.md

```diff
@@ -0,0 +1,681 @@
+  defined.
+- Check that HCs for all load balancers point to `healthCheckNodePort`
+  irrespective of `ExternalTrafficPolicy` `Local` or `Cluster`
+
+###### How can someone using this feature know that it is working for their instance?
+
+- [X] Other (treat as last resort)
+  - Details:
+    - Overall cloud provider/API calls are reduced because of not needing to
+      re-configure load balancers constantly.
+    - Load balancer re-configuration as a cause by transient changes to nodes
+      have zero impact and zero update calls on the load balancer.
+    - Updating a service from `externalTrafficPolicy` `Local` to `Cluster` has
+      little to no impact on ingress SLA.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+TBD, but:
+
+- Updating service from `externalTrafficPolicy` `Local` to `Cluster` should
+  not lead to more than 1 % of new connection errors, and 0 % for established
+  connections.
+- Updating the set of nodes for load balancers should not take more than a
+  couple of seconds until the change is in effect and traffic is 100 % correctly
+  forwarded.
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [X] Metrics
+  - Metric name: The KCCM already has a metric on time taken to update load
+    balancers. This should be reduced and the amount calls in general too.
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+It would be good if Kube-proxy's healthz server exposes a metric for:
+
+- Amount of invocations
+- Time taken to answer to an invocation
+- Return code to invocation
+
+This would help identify how it is doing and why a node might be missing from
+the load balancer set.
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### Does this feature depend on any specific services running in the cluster?
+
+It depends on having a service proxy implement the health check server
+functionality.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+Yes, services of type `externalTrafficPolicy: Cluster` will now get
+`healthCheckNodePort` assigned to them.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+Possibly: we will now end up opening TCP sockets for all services of
```

Comment:
we no longer open TCP sockets to hold open NodePorts

Reply:
we do for HCNP